client/src/components/page-profile.component.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { AppDispatch, RootState } from '../app/store';
+import { getUserData } from '../features/user/userActions';
+
+export const ProfilePageComponent = () => {
+  const dispatch = useDispatch<AppDispatch>();
+  dispatch(getUserData());
+
+  const user = useSelector((state: RootState) => state.user.user);
+
+  if (!user) return <></>;
```

Comment:
Bad practice. Read about chunk loading and it's issues If you do such conditions - you are breaking chunk caching - better to do: `return (<>{user ? (...) : null }</>)`